File: modules/reporting/theme_test.go
// SPDX-FileCopyrightText: 2026 Olivares.AI
// SPDX-License-Identifier: AGPL-3.0-only
// Additional terms under AGPL-3.0-only section 7(a) disclaim warranty and limit liability: see DISCLAIMER.md at the repository root.

package reporting

import "testing"

func TestStatusBadgeClassMapsKnownAndUnknownStatuses(t *testing.T) {
	tests := []struct {
		status string
		want   string
	}{
		{"satisfied", "badge-satisfied"},
		{"by_design", "badge-by-design"},
		{"partial", "badge-partial"},
		{"gap", "badge-gap"},
		{"unmapped", "badge-unmapped"},
		{"unexpected", "badge-unmapped"},
	}
	for _, tt := range tests {
		if got := statusBadgeClass(tt.status); got != tt.want {
			t.Fatalf("statusBadgeClass(%q) = %q, want %q", tt.status, got, tt.want)
		}
	}
}

func TestBudgetBarClassMapsThresholdsAndOverBudget(t *testing.T) {
	tests := []struct {
		pct  int
		over bool
		want string
	}{
		{20, false, "progress-ok"},
		{80, false, "progress-ok"},
		{81, false, "progress-warn"},
		{10, true, "progress-over"},
	}
	for _, tt := range tests {
		if got := budgetBarClass(tt.pct, tt.over); got != tt.want {
			t.Fatalf("budgetBarClass(%d,%v) = %q, want %q", tt.pct, tt.over, got, tt.want)
		}
	}
}

File: web/src/stores/theme.test.ts
// SPDX-FileCopyrightText: 2026 Olivares.AI
// SPDX-License-Identifier: AGPL-3.0-only
// Additional terms under AGPL-3.0-only section 7(a) disclaim warranty and limit liability: see DISCLAIMER.md at the repository root.
import { beforeEach, describe, expect, it } from 'vitest'
import { resolveDark, useThemeStore } from './theme'

beforeEach(() => {
  localStorage.clear()
  document.documentElement.classList.remove('dark')
})

describe('theme store', () => {
  it('applies the dark class and persists the raw key on setTheme', () => {
    useThemeStore.getState().setTheme('dark')
    expect(document.documentElement.classList.contains('dark')).toBe(true)
    expect(localStorage.getItem('olivares.theme')).toBe('dark')
    expect(useThemeStore.getState().resolved).toBe('dark')

    useThemeStore.getState().setTheme('light')
    expect(document.documentElement.classList.contains('dark')).toBe(false)
    expect(localStorage.getItem('olivares.theme')).toBe('light')
    expect(useThemeStore.getState().resolved).toBe('light')
  })

  it('persists the raw string under the same key the no-FOUC bootstrap reads', () => {
    useThemeStore.getState().setTheme('system')
    // NOT a JSON-wrapped value — index.html reads it verbatim before any JS bundle.
    expect(localStorage.getItem('olivares.theme')).toBe('system')
  })

  it('resolveDark mirrors the bootstrap logic', () => {
    expect(resolveDark('dark')).toBe(true)
    expect(resolveDark('light')).toBe(false)
    // 'system' depends on matchMedia, stubbed to no-match in tests → light.
    expect(resolveDark('system')).toBe(false)
  })
})
